File: sem3_csharp-master/sem3_csharp-master/AspnetApi/Models/Product.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using System.Drawing.Drawing2D;

namespace AspnetApi.Models
{
    public class Product
    {

        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }

        public float Price { get; set; }

        public float SalePrice { get; set; }

        public bool? Active { get; set; }

        public string? Image { get; set; }

        [Required]
        public string ProductName { get; set; }

        public string? Description { get; set; }

        public string Slug { get; set; }


        [ForeignKey("Category")]
        public int CategoryId { get; set; }
        public Category Category { get; set; }

        public string? Album { get; set; }

        public DateTime? CreatedAt { get; set; } = DateTime.Now;

        public DateTime? UpdatedAt { get; set; }

        [MaxLength(100)]
        public string? CreatedBy { get; set; }

        [MaxLength(100)]
        public string? UpdatedBy { get; set; }
        public bool? DeletedAt { get; set; }
    }
}

File: customer/lib/models/product.dart
class Product {
  final int id;
  final String name;
  final double price;
  final String image;
  final String description;// đường dẫn tới ảnh trong assets

  Product({
    required this.id,
    required this.name,
    required this.price,
    required this.image,
    required this.description,

  });
}

// Giỏ hàng (tạm lưu global)
List<Product> cart = [];
